cache/blobs.go

```diff
@@ -44,29 +45,19 @@ func (sr *immutableRef) computeBlobChain(ctx context.Context, createIfNeeded boo
 }
 
 func computeBlobChain(ctx context.Context, sr *immutableRef, createIfNeeded bool, compressionType compression.Type, forceCompression bool, s session.Group) error {
-	baseCtx := ctx
 	eg, ctx := errgroup.WithContext(ctx)
-	var currentDescr ocispecs.Descriptor
 	if sr.parent != nil {
 		eg.Go(func() error {
 			return computeBlobChain(ctx, sr.parent, createIfNeeded, compressionType, forceCompression, s)
 		})
 	}
+
 	eg.Go(func() error {
-		dp, err := g.Do(ctx, sr.ID(), func(ctx context.Context) (interface{}, error) {
-			refInfo := sr.Info()
-			if refInfo.Blob != "" {
-				if forceCompression {
-					desc, err := sr.ociDesc()
-					if err != nil {
-						return nil, err
-					}
-					if err := ensureCompression(ctx, sr, desc, compressionType, s); err != nil {
-						return nil, err
-					}
-				}
-				return nil, nil
-			} else if !createIfNeeded {
+		v, err := g.Do(ctx, fmt.Sprintf("%s-%t", sr.ID(), createIfNeeded), func(ctx context.Context) (interface{}, error) {
```

Comment:
doesn't this skip the second mediatype or am I missing something. The call to `ensureCompression` looks to be in this block, therefore never called for the second attempt.

Reply:
The call to `ensureCompression` is outside of the flightcontrol call, so multiple calls for the same blob w/ different mediatypes should still each call `ensureCompression` with their mediatype.